File: CS-Examples/26_StructuredDocumentTag/UpdateCheckBox.cs
using System;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;
using System.Collections.Generic;

namespace UpdateCheckBox
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
         
            // Create a new document object
            Document document = new Document();

            // Load a document from the specified file path
            document.LoadFromFile(@"..\..\..\..\..\..\Data\CheckBoxContentControl.docx");

            // Get all the StructureTags from the document
            StructureTags structureTags = GetAllTags(document);

            // Get the list of StructureDocumentTagInline objects from the StructureTags
            List<StructureDocumentTagInline> tagInlines = structureTags.tagInlines;

            // Iterate through the list of StructureDocumentTagInline objects
            for (int i = 0; i < tagInlines.Count; i++)
            {
                // Get the SDTType of the current StructureDocumentTagInline
                string type = tagInlines[i].SDTProperties.SDTType.ToString();

                // Check if the SDTType is "CheckBox"
                if (type == "CheckBox")
                {
                    // Get the SdtCheckBox from the ControlProperties of the StructureDocumentTagInline
                    SdtCheckBox scb = tagInlines[i].SDTProperties.ControlProperties as SdtCheckBox;

                    // Toggle the Checked property of the SdtCheckBox
                    if (scb.Checked)
                    {
                        scb.Checked = false;
                    }
                    else
                    {
                        scb.Checked = true;
                    }
                }
            }

            // Save the modified document to "Output.docx" in DOCX format
            document.SaveToFile("Output.docx", FileFormat.Docx);

            // Dispose the document object
            document.Dispose();

            //Launch the Word file.
            WordDocViewer("Output.docx");

        }

        // Define a method named "GetAllTags" that takes a Document object as input and returns a StructureTags object
        static StructureTags GetAllTags(Document document)
        {
            // Create a new StructureTags object to store the StructureDocumentTagInline objects
            StructureTags structureTags = new StructureTags();

            // Iterate through the sections in the document
            foreach (Section section in document.Sections)
            {
                // Iterate through the child objects in the section's body
                foreach (DocumentObject obj in section.Body.ChildObjects)
                {
                    // Check if the current object is a Paragraph
                    if (obj.DocumentObjectType == DocumentObjectType.Paragraph)
                    {
                        // Iterate through the child objects in the paragraph
                        foreach (DocumentObject pobj in (obj as Paragraph).ChildObjects)
                        {
                            // Check if the current object is a StructureDocumentTagInline
                            if (pobj.DocumentObjectType == DocumentObjectType.StructureDocumentTagInline)
                            {
                                // Add the StructureDocumentTagInline to the tagInlines list in the StructureTags object
                                structureTags.tagInlines.Add(pobj as StructureDocumentTagInline);
                            }
                        }
                    }
                }
            }

            // Return the StructureTags object containing the collected StructureDocumentTagInline objects
            return structureTags;
        }

        // Define a public class named "StructureTags"
        public class StructureTags
        {
            List<StructureDocumentTagInline> m_tagInlines;
            public List<StructureDocumentTagInline> tagInlines
            {
                get
                {
                    if (m_tagInlines == null)
                        m_tagInlines = new List<StructureDocumentTagInline>();
                    return m_tagInlines;
                }
                set
                {
                    m_tagInlines = value;
                }
            }
        }
        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB-Examples/26_StructuredDocumentTag/UpdateCheckBox.vb
Imports Spire.Doc
Imports Spire.Doc.Documents

Namespace UpdateCheckBox
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new document object
			Dim document As New Document()

			' Load a document from the specified file path
			document.LoadFromFile("..\..\..\..\..\..\Data\CheckBoxContentControl.docx")

			' Get all the StructureTags from the document
			Dim structureTags As StructureTags = GetAllTags(document)

			' Get the list of StructureDocumentTagInline objects from the StructureTags
			Dim tagInlines As List(Of StructureDocumentTagInline) = structureTags.tagInlines

			' Iterate through the list of StructureDocumentTagInline objects
			For i As Integer = 0 To tagInlines.Count - 1
				' Get the SDTType of the current StructureDocumentTagInline
				Dim type As String = tagInlines(i).SDTProperties.SDTType.ToString()

				' Check if the SDTType is "CheckBox"
				If type = "CheckBox" Then
					' Get the SdtCheckBox from the ControlProperties of the StructureDocumentTagInline
					Dim scb As SdtCheckBox = TryCast(tagInlines(i).SDTProperties.ControlProperties, SdtCheckBox)

					' Toggle the Checked property of the SdtCheckBox
					If scb.Checked Then
						scb.Checked = False
					Else
						scb.Checked = True
					End If
				End If
			Next i

			' Save the modified document to "Output.docx" in DOCX format
			document.SaveToFile("Output.docx", FileFormat.Docx)

			' Dispose the document object
			document.Dispose()

			'Launch the Word file.
			WordDocViewer("Output.docx")

		End Sub

		' Define a method named "GetAllTags" that takes a Document object as input and returns a StructureTags object
		Private Shared Function GetAllTags(ByVal document As Document) As StructureTags
			' Create a new StructureTags object to store the StructureDocumentTagInline objects
			Dim structureTags As New StructureTags()

			' Iterate through the sections in the document
			For Each section As Section In document.Sections
				' Iterate through the child objects in the section's body
				For Each obj As DocumentObject In section.Body.ChildObjects
					' Check if the current object is a Paragraph
					If obj.DocumentObjectType = DocumentObjectType.Paragraph Then
						' Iterate through the child objects in the paragraph
						For Each pobj As DocumentObject In (TryCast(obj, Paragraph)).ChildObjects
							' Check if the current object is a StructureDocumentTagInline
							If pobj.DocumentObjectType = DocumentObjectType.StructureDocumentTagInline Then
								' Add the StructureDocumentTagInline to the tagInlines list in the StructureTags object
								structureTags.tagInlines.Add(TryCast(pobj, StructureDocumentTagInline))
							End If
						Next pobj
					End If
				Next obj
			Next section

			' Return the StructureTags object containing the collected StructureDocumentTagInline objects
			Return structureTags
		End Function

		' Define a public class named "StructureTags"
		Public Class StructureTags
			Private m_tagInlines As List(Of StructureDocumentTagInline)
			Public Property tagInlines() As List(Of StructureDocumentTagInline)
				Get
					If m_tagInlines Is Nothing Then
						m_tagInlines = New List(Of StructureDocumentTagInline)()
					End If
					Return m_tagInlines
				End Get
				Set(ByVal value As List(Of StructureDocumentTagInline))
					m_tagInlines = value
				End Set
			End Property
		End Class
		
		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
